File: 169. Majority Element/169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # O(nlogn)
        # nums.sort()
        # return nums[len(nums)//2]

        # O(n)
        d = dict()
        for num in nums:
            d[num] = d[num]+1 if num in d else 1
            if d[num] > len(nums)//2:
                return num

File: 169. Majority Element/169.js
/**
 
@param {number[]} nums
@return {number}*/
var majorityElement = function(nums) {
    const hash = {};
    const length = nums.length / 2
    for (let i = 0; i < nums.length; i++) {
        if (hash[nums[i]] === undefined) {
            hash[nums[i]] = 1;
        } else {
            hash[nums[i]]++;
        }

        if (hash[nums[i]] > length) {
            return nums[i]
        } 
    }
};